basics/uattributes.adoc

```diff
@@ -96,13 +124,31 @@ The following table defines attributes that are used for publish messages in add
 
 |`type`
 |yes
-|*MUST* be set to `UMESSAGE_TYPE_PUBLISH`.
+a| 
+[.specitem,oft-sid="req~up-attributes-publish-type~1",oft-needs="impl,utest"]
+--
+*MUST* be set to `UMESSAGE_TYPE_PUBLISH`.
+--
 
 |`source`
 |yes
 a|A link:uri.adoc[UUri] representing the topic that this message is published to.
 
+[.specitem,oft-sid="req~up-attributes-publish-source~1",oft-needs="impl,utest"]
+--
 * The UUri's `resource_id` *MUST* be set to a value in range `[0x8000, 0xFFFE]`.
+--
+
+|`sink`
+|no
+a|A link:uri.adoc[UUri] representing the receiver of this published message. This information is 
+only required when we are sending the published event between devices through the streamer, publishers are not required to populate this information.
```

Comment:
```suggestion
only required when we are sending the published event between devices through a streamer. A uEntity sending a Publish message is not required to populate this information.
```

Clarification ideas to separate a typical uE sending a publish message vs a streamer sending a publish message